File: Chapter 11/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter11/section_11_4_secrets_management/hardcoded/MockDatabaseConnection.java
package com.grokkingsoftwarearchitecture.chapter11.section_11_4_secrets_management.hardcoded;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * MOCK DATABASE CONNECTION - FOR DEMONSTRATION ONLY
 * 
 * This simulates a database connection to show the hardcoded credentials problem.
 * In reality, this would be a JDBC/PostgreSQL connection or similar.
 */
public class MockDatabaseConnection {
    private final String host;
    private final String database;
    private final String user;
    private final String password;
    private boolean isConnected;

    public MockDatabaseConnection(String host, String database, String user, String password) {
        this.host = host;
        this.database = database;
        this.user = user;
        this.password = password;
        this.isConnected = false;
    }

    public boolean connect() {
        // Simulate connection logic
        if ("admin".equals(user) && "Password123!".equals(password)) {
            isConnected = true;
            return true;
        } else {
            throw new IllegalStateException("Authentication failed for user " + user);
        }
    }

    public List<Map<String, Object>> executeQuery(String query) {
        if (!isConnected) {
            throw new IllegalStateException("Not connected to database");
        }

        // Return mock data
        List<Map<String, Object>> orders = new ArrayList<>();
        Map<String, Object> order = new HashMap<>();
        order.put("order_id", "ORD-001");
        order.put("customer_id", "CUST-123");
        order.put("total", 99.99);
        orders.add(order);
        
        return orders;
    }

    public void close() {
        isConnected = false;
    }
}

File: Chapter 11/C#/Section_11_4_Secrets_Management/Hardcoded/MockDatabaseConnection.cs
namespace Chapter11.Hardcoded;

/// <summary>
/// MOCK DATABASE CONNECTION - FOR DEMONSTRATION ONLY
/// 
/// This simulates a database connection to show the hardcoded credentials problem.
/// In reality, this would be NpgsqlConnection or similar.
/// </summary>
public class MockDatabaseConnection
{
    public string Host { get; }
    public string Database { get; }
    public string User { get; }
    public string Password { get; }
    private bool _isConnected;

    public MockDatabaseConnection(string host, string database, string user, string password)
    {
        Host = host;
        Database = database;
        User = user;
        Password = password;
        _isConnected = false;
    }

    public bool Connect()
    {
        // Simulate connection logic
        if (User == "admin" && Password == "Password123!")
        {
            _isConnected = true;
            return true;
        }
        else
        {
            throw new InvalidOperationException($"Authentication failed for user {User}");
        }
    }

    public List<Dictionary<string, object>> ExecuteQuery(string query)
    {
        if (!_isConnected)
        {
            throw new InvalidOperationException("Not connected to database");
        }

        // Return mock data
        return new List<Dictionary<string, object>>
        {
            new Dictionary<string, object>
            {
                { "order_id", "ORD-001" },
                { "customer_id", "CUST-123" },
                { "total", 99.99 }
            }
        };
    }

    public void Close()
    {
        _isConnected = false;
    }
}